packages/shylu/shylu_dd/frosch/src/Adapters/Thyra_FROSchFactory_def.hpp

```diff
@@ -156,26 +211,89 @@ namespace Thyra {
             } else {
                 FROSCH_ASSERT(false,"ERROR: Specify a valid DofOrdering.");
             }
+
             RCP<GDSWPreconditioner<SC,LO,GO,NO> > GP(new GDSWPreconditioner<SC,LO,GO,NO>(A,paramList_));
-            GP->initialize(paramList_->get("Dimension",3),paramList_->get("DofsPerNode",1),dofOrdering,paramList_->get("Overlap",1),repeatedMap,coordinatesList);
+
+            GP->initialize(paramList_->get("Dimension",3),
+                           paramList_->get("DofsPerNode",1),
+                           dofOrdering,
+                           paramList_->get("Overlap",1),
+                           repeatedMap,
+                           coordinatesList);
+
             SchwarzPreconditioner = GP;
         } else if (!paramList_->get("FROSch Preconditioner Type","TwoLevelPreconditioner").compare("OneLevelPreconditioner")) {
+            // Extract the repeated map
             ConstMapPtr repeatedMap = Teuchos::null;
             if (paramList_->isParameter("Repeated Map")) {
-                repeatedMap = ExtractPtrFromParameterList<ConstMap>(*paramList_,"Repeated Map");
+                repeatedMap = ExtractPtrFromParameterList<Map>(*paramList_,"Repeated Map").getConst();
```

Comment:
It looks like you repeat this chunk of code a few times -- would you consider factoring it out into a function?

Reply:
I put those parts into separate functions.